execute/plugin_test.go

```diff
@@ -166,3 +155,496 @@ func Test_getPendingExecutedReports(t *testing.T) {
+	if _, err := rand.Read(bytes); err != nil {
+		panic(err)
+	}
+	return hex.EncodeToString(bytes)[:32]
+}
+
+func breakCommitReport(
+	commitReport cciptypes.ExecutePluginCommitDataWithMessages,
+) cciptypes.ExecutePluginCommitDataWithMessages {
+	commitReport.Messages = append(commitReport.Messages, cciptypes.CCIPMsg{
+		CCIPMsgBaseDetails: cciptypes.CCIPMsgBaseDetails{
+			ID:          mustRandID(),
+			SourceChain: cciptypes.ChainSelector(1),
+			SeqNum:      cciptypes.SeqNum(999),
+			MsgHash:     cciptypes.Bytes32{},
+		},
+	})
+	return commitReport
+}
+
+func makeTestCommitReport(
+	numMessages, srcChain, firstSeqNum, block int, timestamp int64, executed []cciptypes.SeqNum,
+) cciptypes.ExecutePluginCommitDataWithMessages {
+	for _, e := range executed {
+		if e < cciptypes.SeqNum(firstSeqNum) || e > cciptypes.SeqNum(firstSeqNum+numMessages-1) {
+			panic("executed message out of range")
+		}
+	}
+	var messages []cciptypes.CCIPMsg
+	for i := 0; i < numMessages; i++ {
+		messages = append(messages, cciptypes.CCIPMsg{
+			CCIPMsgBaseDetails: cciptypes.CCIPMsgBaseDetails{
+				ID:          mustRandID(),
+				SourceChain: cciptypes.ChainSelector(srcChain),
+				SeqNum:      cciptypes.SeqNum(i + firstSeqNum),
+				MsgHash:     cciptypes.Bytes32{},
+			},
+			Nonce: uint64(i),
+		})
+	}
+
+	sequenceNumberRange :=
+		cciptypes.NewSeqNumRange(cciptypes.SeqNum(firstSeqNum), cciptypes.SeqNum(firstSeqNum+numMessages-1))
+	return cciptypes.ExecutePluginCommitDataWithMessages{
+		ExecutePluginCommitData: cciptypes.ExecutePluginCommitData{
+			SourceChain:         cciptypes.ChainSelector(srcChain),
+			SequenceNumberRange: sequenceNumberRange,
+			Timestamp:           time.UnixMilli(timestamp),
+			BlockNum:            uint64(block),
+			ExecutedMessages:    executed,
+		},
+		Messages: messages,
+	}
+
+}
+
+// assertMerkleRoot computes the source messages merkle root, then computes a verification with the proof, then compares
+// the roots.
+func assertMerkleRoot(
+	t *testing.T,
+	hasher cciptypes.MessageHasher,
+	execReport cciptypes.ExecutePluginReportSingleChain,
+	commitReport cciptypes.ExecutePluginCommitDataWithMessages,
+) {
+	keccak := hashutil.NewKeccak()
+	// Generate merkle root from commit report messages
+	var leafHashes [][32]byte
+	for _, msg := range commitReport.Messages {
+		hash, err := hasher.Hash(context.Background(), msg)
+		require.NoError(t, err)
+		leafHashes = append(leafHashes, hash)
+	}
+	tree, err := merklemulti.NewTree(keccak, leafHashes)
+	require.NoError(t, err)
+	merkleRoot := tree.Root()
+
+	// Generate merkle root from exec report messages and proofj
+	ctx := context.Background()
+	var leaves [][32]byte
+	for _, msg := range execReport.Messages {
+		hash, err := hasher.Hash(ctx, msg)
+		require.NoError(t, err)
+		leaves = append(leaves, hash)
+	}
+	proofCast := make([][32]byte, len(execReport.Proofs))
+	for i, p := range execReport.Proofs {
+		copy(proofCast[i][:], p[:32])
+	}
+	var proof merklemulti.Proof[[32]byte]
```

Comment:
Ah this is interesting, the generics won't work with the `cciptypes.Bytes32` type?